src/components/OptionsMenu.js

```diff
@@ -0,0 +1,190 @@
+ *          onPress: () => {},
+ *          },
+ *        }]
+ *        renderDefault={false}
+ *      />
+ *      <Button
+ *        width={200}
+ *        width={200}
+ *        label="Show Options"
+ *        onPress={() => {
+ *        setShowOptions(true);
+ *        }}
+ *      />
+ *    </Container>
+ *  );
+ * }
+ * ```
+ *
+ */
+
+const RenderOptions = ({
+  containerProps,
+  onPress,
+  iconName,
+  title,
+  labelProps,
+  variant,
+}) => {
+  const theme = useContext(ThemeContext);
+
+  return (
+    <Touchable onPress={onPress}>
+      <Container
+        flexDirection="row"
+        alignItems="center"
+        py={10}
+        {...containerProps}
+      >
+        <Icon
+          size="20"
+          color={
+            variant === "danger"
+              ? theme.colors.font.danger500
+              : labelProps?.color ?? theme.colors.font.grey800
+          }
+          name={iconName}
+        />
+        <Typography
+          fontFamily="sf400"
+          color={
+            variant === "danger" ? theme.colors.font.danger500 : "font.grey800"
+          }
+          fontSize="l"
+          ml={14}
+          {...labelProps}
+        >
+          {title}
+        </Typography>
+      </Container>
+    </Touchable>
+  );
+};
+
+export const OptionsMenu = ({
+  isVisible,
+  renderDefault,
+  onEdit,
+  onDelete,
+  hide,
+  options,
+  containerProps,
+}) => {
+  const defaultOptions =
+    renderDefault || !options?.length
+      ? [
+          {
+            title: "Edit",
+            iconName: "edit-2-line",
+            onPress: onEdit,
+          },
+          {
+            title: "Remove",
+            iconName: "delete-bin-4-line",
+            onPress: onDelete,
+            variant: "danger",
+          },
+        ]
+      : [];
+
+  return (
+    <Container>
+      <BottomSheet isVisible={isVisible} hide={hide}>
+        {[...defaultOptions, ...options].map(
+          ({ iconName, labelProps, onPress, title, variant }, index) => (
+            <RenderOptions
+              key={index}
+              iconName={iconName}
+              labelProps={labelProps}
+              onPress={onPress}
+              title={title}
+              containerProps={containerProps}
+              variant={variant}
+            />
+          )
+        )}
+      </BottomSheet>
+    </Container>
+  );
+};
+
+OptionsMenu.defaultProps = {
+  isVisible: false,
+  renderDefault: true,
+  onEdit: () => {},
+  onDelete: () => {},
+  hide: () => {},
+  options: [],
+  containerProps: {},
+};
+
+RenderOptions.propTypes = {
+  onPress: PropTypes.func,
+  iconName: PropTypes.string,
+  title: PropTypes.string,
+  labelProps: PropTypes.object,
+  containerProps: PropTypes.object,
+  variant: PropTypes.oneOf(["default", "danger"]),
+};
+
+RenderOptions.defaultOptions = {
+  variant: "default",
+};
+
+OptionsMenu.propTypes = {
+  isVisible: PropTypes.bool,
+  renderDefault: PropTypes.bool,
+  onEdit: PropTypes.func,
```

Comment:
Instead of just setting the PropTypes as func, we can write a custom function which would give us a warning if those are not passed while integrating the component, also will give us an insight on how these props are interdependent on each other. 
```
onEdit: function (props, propName) {
    if (
      (props["renderDefault"] === true || !props["options"]?.length) &&
      (props[propName] == undefined || typeof props[propName] != "function")
    ) {
      return new Error("Please provide onEdit function!");
    }
  },
  onDelete: function (props, propName) {
    if (
      (props["renderDefault"] === true || !props["options"]?.length) &&
      (props[propName] == undefined || typeof props[propName] != "function")
    ) {
      return new Error("Please provide onDelete function!");
    }
  },
```
With the current approach we won't get any warning if we are not passing the onEdit func, also to know when we need to pass these props we need to go through the code completely inorder to understand the dependencies with other props. But if we write custom prop then just going through the condition in the propType will give us an idea when that prop is used.
 
 @bigbinary/neeto-rn Please share your thoughts on this, if adding custom propType functions for nested props would be good or not.